test/cmdlineTests/standard_metadata_custom_cleanup_sequence/input.json

```diff
@@ -0,0 +1,23 @@
+{
+	"language": "Solidity",
+	"sources": {
+		"C": {"urls": ["standard_metadata/in.sol"]}
+	},
+	"settings": {
+		"outputSelection": {
+			"*": {"*": ["metadata"]}
+		},
+		"viaIR": true,
+		"optimizer": {
+			"enabled": true,
+			"details": {
+				"yul": true,
+				"yulDetails": {
+					// default optimization sequence with a custom (truncated) cleanup sequence.
+					// 0.8.18 - 0.8.28 this would spuriously be interpreted as defaulted optimizer settings in metadata.
+					"optimizerSteps": "dhfoDgvulfnTUtnIf[xa[r]EscLMcCTUtTOntnfDIulLculVcul [j]Tpeulxa[rul]xa[r]cLgvifCTUca[r]LSsTFOtfDnca[r]Iulc]jmul[jul] VcTOcul jmul:D"
```

Comment:
I don't like this test so much actually. We won't regress if the default sequence changes at some point and it goes unnoticed to update it here as well.

Reply:
Another option would be to add this case to boost-based metadata tests: https://github.com/ethereum/solidity/blob/9005353d7361f153701d5fc816c1af6e1d436bcd/test/libsolidity/Metadata.cpp#L452-L459

There you can refer to `OptimiserSettings` and the default sequence without hard-coding it.

Reply:
Nice, yeah, much better! I had to refactor it a bit because it was testing against `minimal` settings plus yul optimizations which was already caught by the old comparison operator. For the regression to kick in, I now take `standard` settings and modify only the cleanup sequence.

Reply:
Maybe let's just make it loop over all presets (`OptimisationPreset`)? This bug shows that we can't really assume that this is completely independent of other settings so this would give us a bit more coverage and also make the test more regular.

For completness, we should also have a test that non-default main sequence and default cleanup works.

We also don't really test that `minimal` translates to just `enabled: false`, `standard` to `enabled: false` and other presets to different things and we should.

Reply:
Increased the coverage accordingly.